### PR TITLE
feature: authenticate ftx websocket

### PR DESCRIPTION
**Pull request by ycdesu**
If you don't invoke `SetPublicOnly`,  I will login when connecting. Note that I haven't subscribe orders and trades yet, so the `orderupdate` command doesn't work yet.

### CODE REVIEW

File: pkg/cmd/orderbook.go [hunk 1]
```diff
@@ -62,9 +66,76 @@ var orderbookCmd = &cobra.Command{
 	},
 }
 
+// go run ./cmd/bbgo orderupdate
```
Comment (c9s) [inline pkg/cmd/orderbook.go:69]: do we really need this command? looks like only for testing and development

Comment (ycdesu) [reply pkg/cmd/orderbook.go:69]: Yes it's for testing only. Do I have to remove them?

Comment (c9s) [reply pkg/cmd/orderbook.go:69]: move to examples?

File: pkg/exchange/ftx/stream.go [hunk 2]
```diff
@@ -28,18 +29,39 @@ func NewStream(key, secret string) *Stream {
 }
 
 func (s *Stream) Connect(ctx context.Context) error {
-	return s.wsService.Connect(ctx)
+	// If it's not public only, let's do the authentication.
+	if atomic.LoadInt32(&s.publicOnly) == 0 {
+		logger.Infof("subscribe private events")
```
Comment (c9s) [inline pkg/exchange/ftx/stream.go:34]: remove debug log?

File: pkg/exchange/ftx/stream.go [hunk 3]
```diff
@@ -28,18 +29,39 @@ func NewStream(key, secret string) *Stream {
...
 
 func (s *Stream) SetPublicOnly() {
 	atomic.StoreInt32(&s.publicOnly, 1)
 }
 
 func (s *Stream) Subscribe(channel types.Channel, symbol string, _ types.SubscribeOptions) {
-	if err := s.wsService.Subscribe(channel, symbol); err != nil {
-		logger.WithError(err).Errorf("subscribe failed, should never happen")
+	if channel != types.BookChannel {
+		// TODO: return err
```
Comment (c9s) [inline pkg/exchange/ftx/stream.go:53]: you can panic here, since it's logical exception

**Review by c9s — APPROVED** [review]

**Pull request merged**